crc32_field.go

```diff
@@ -6,9 +6,17 @@ import (
 	"hash/crc32"
 )
 
+const (
+	crcIEEE = iota
+	crcCastagnoli
```

Comment:
Why define these? you can just have `newCRC32Field` take the `crc32`-defined constant directly?

Reply:
Do you mean I should use the polynomial constants directly? I could, as they are just ints, but it didn't seem very clean to me as the intent here is to represent an enumeration of supported polynomials in the context of Kafka protocol, not the polynomial itself.
But if you feel strongly about it, I can remove them.

Reply:
That's what I meant, but I see your point. In that case it probably makes sense to explicitly type it (e.g. the way that compression codecs are) rather than using ints, since the polynomials themselves are just ints.